tests/TestCase/EntityProxyTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Algolia\SearchBundle\TestCase;
+
+use Algolia\SearchBundle\BaseTest;
+use Algolia\SearchBundle\TestApp\Entity\Comment;
+use Algolia\SearchBundle\Util\ClassInfo;
+
+class EntityProxyTest extends BaseTest
+{
+    private static $values = [];
+
+    public static function setUpBeforeClass(): void
+    {
+        // Unset env variables to make sure Algolia
+        // Credentials are only required when the
+        // client is used. Save them to restore them after.
+        // See: https://github.com/algolia/search-bundle/issues/241
+        self::$values = [
+            'env_id'  => getenv('ALGOLIA_APP_ID'),
+            'env_key' => getenv('ALGOLIA_API_KEY'),
+            '_env'    => $_ENV,
+            '_server' => $_SERVER,
```

Comment:
Can we store only the `$_ENV` and `$_SERVER` keys that we unset, please? It doesn't seem super safe to me to keep the entire array in memory when we don't have to.